wsclient/src/main/java/org/hpccsystems/ws/client/HPCCWsDFUClient.java

```diff
@@ -888,7 +890,26 @@ public void deleteSuperFileSubfiles(String superfile, List<String> subfiles) thr
 
         request.setType(DFUArrayActions.Factory.fromValue("Delete"));
         EspStringArray logicalfilesarray = new EspStringArray();
-        logicalfilesarray.setItem(files.toArray(new String[files.size()]));
+        String[] filesarray = files.toArray(new String[files.size()]);
+
+        if (cluster != null && !cluster.isEmpty())
+        {
+            for(int i = 0; i < filesarray.length; i++)
+            {
+                String fullfile = filesarray[i];
+                String[] split = fullfile.split("::");
+                String file = split[split.length-1];
+
+                int atIndex = file.indexOf('@');
```

Comment:
You probably already mentioned this, but can files have the '@' symbol in their name? If so would this create false positives for those files?

Reply:
Good question, but from documentation and a few tests, the @ symbol can only be used as a delimiter between filename and groupname. the odd thing is that the scopes can have the @ symbol

Reply:
Gotcha looks good to me